src/main/java/ma/boot/springboot/repository/WordRepository.java

```diff
@@ -0,0 +1,17 @@
+package ma.boot.springboot.repository;
+
+import java.util.List;
+import ma.boot.springboot.model.Word;
+import ma.boot.springboot.model.dto.WordResponseDto;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface WordRepository extends JpaRepository<Word, Long> {
+    @Query("SELECT "
+            + "new ma.boot.springboot.model.dto.WordResponseDto(w.theWord) "
+            + " from Word w "
+            + " Group by w.theWord"
+            + " ORDER BY COUNT(w.theWord) DESC, w.theWord ASC")
```

Comment:
I am not sure that your query works correctly
I would rather suggest to add `quantity` field to Word model and save words already with this quantity to the db

Reply:
it works all right - I have checked it out. My table will be huge and this is downside of my solution...
certainly I can try to do stream and have WORD class with quantity/occurence field - I thought to avoid complex "stream in stream"...